File: app/templates/server_go/gin/api/todo/controller/todocontroller.go
package todocontroller

import (
	"<%= repoHostUrl %>/<%= userNameSpace %>/<%= appName %>/server/api/todo/dao"
	todo "<%= repoHostUrl %>/<%= userNameSpace %>/<%= appName %>/server/api/todo/model"
	"github.com/gin-gonic/gin"
	"net/http"
)

func GetAll(c *gin.Context) error {
	ts, _ := tododao.All()

	return c.JSON(http.StatusOK, ts)
}

func NewTodo(c *gin.Context) error {
	t := new(todo.Todo)

	c.Bind(t)

	nt, _ := tododao.NewTodo(*t)

	return c.JSON(http.StatusOK, nt)
}

func RemoveTodo(c *gin.Context) error {
	id := c.Param("id")

	tododao.DeleteTodo(id)

	return c.String(http.StatusOK, "")
}

File: app/templates/client/ng1/dev/todo/controllers/todo-controller.js
;(function(ng) {
  "use strict";

  ng.module("<%= appName %>")
    .controller("TodoController", [
      "$log",
      "Todo",
      "TodoDAO",
      function($log, Todo, TodoDAO) {
        this.todo = new Todo();
        this.todos = [];
        this.title = "Angular1 Todo";
        
        this.createTodo = function(todo) {
          TodoDAO.createTodo(todo)
            .then((newTodo) => {
              this.todos.push(newTodo);
              this.todo = new Todo();
            })
            .catch($log.error);
        };
        
        this.deleteTodo = function(id) {
          TodoDAO.deleteTodo(id)
            .then(() => {
              return TodoDAO.getAll()
                .then((todos) => {
                  return this.todos = todos;
                });
            })
            .catch($log.error);
        };

        ;(() => {
          return TodoDAO.getAll()
            .then((todos) => {
              return this.todos = todos;
            })
            .catch($log.error);
        })();
      }
    ]);
}(window.angular));
